### ISSUE TITLE
Zijn Output, Input en Rule (uit CPSV) kernobjecten?

### ISSUE THREAD

**Issue by DieterDePaepe** [Informatievlaanderen/OSLO#12]
Moeten deze concepten apart in registers worden bijgehouden?

**Comment by dhaeneth** [comment]
Ja. Zowel input als output vormen instanties van het hogere concept 'Product' en geven eigenlijk de opening om bvb. procesarchitectuur en informatie-objecten mee te nemen in het verhaal van dienstverlening. Of het fysiek een afzonderlijk register moet zijn (of meegenomen dient te worden in het register Dienstverlening) is ter discussie. Logisch gezien zijn het wel afzonderlijke objecten.

Ook Rule vormt een afzonderlijk object met een eigen cyclus. Dit is eigenlijk regelgeving die willen capteren (ter inspiratie hebben we hier het LBLOD-project).

**Comment by GeertThijs** [comment]
Zie ook #7. Waar geargumenteerd wordt dat het geen kwestie is van gaan we die entiteiten al dan niet in registers bijhouden (want dat is een beslissing waarop later soms wordt terug gekomen). Het gaat er hem om of die entiteiten als zelfstandig object op zich kunnen bestaan zonder af te hangen van een andere entiteit.

**Comment by GeertThijs** [comment]
@dhaeneth: Oké dus voor Input & Output. Maar hoe zit het met Rule. Momenteel heb ik er in de refactored CPSV-AP een datatype van gemaakt in de veronderstelling dat een Rule niet kan bestaan zonder de PublicService die hij ondersteunt (het is maw een kenmerk van de PublicService). In het geval vd PublicService "ToelatingVoorDeRommelmarkt" gaat het om het gemeentelijk "ReglementVoorHetHoudenVanEenRommelmarkt" dat een implementatie is van de "WetOpDeAmbulazteHandel". Vraag is dus of jij voorbeelden kent (bv uit LBLOD) van Rules die volledig los staan van een PublicService. Definitie in CPSV is "The Rule class represents a document that sets out the specific rules, guidelines or procedures
that the Public Service follows. It includes the terms of service, licence, and authentication
requirements of the service.". Is dus echt wel gedefinieerd vanuit het standpunt van de PublicService, het lijkt niet zonder te kunnen bestaan.

**Comment by GeraldineNolf** [comment]
Zie ook #35 voor het verschil tussen een "product" (data, dienst, document) en zijn metadatabeschrijving vs een "dienstverlening". 
Uiteraard kunnen die producten in registers voorkomen. We doen dit nu al: bv. in het AGIV- en Geopunt-metadatacenter als back-office voor Geopunt voor de geo-wereld (ISO/INSPIRE-standaarden); in de back-office van het Open data portaal van de Vlaamse overheid (CKAN en DCAT-AP-standaarden). In de toekomst worden deze + de pijler rond Documenten gebundeld in de Informatiecatalogus. Moet dé gouden gids aan gegevens en informatie worden van de Overheidsinstanties in Vlaanderen. We hebben nog een lange weg te gaan...

**Comment by GeertThijs** [comment]
@GeraldineNolf: Ja, er ontbreken zeker nog wel wat registers. Of zijn het catalogen? En is er nu een verschil of niet tussen een register en een cataloog en een metadatacataloog? Een register waarin elke Dienstverlening authentiek wordt geregistreerd met bijbehorende beschrijvende gegevens. Versus een cataloog die zegt welke dienstverleningen er allemaal te vinden zijn (Gouden Gids van dienstverleningen) of ze nu authentiek geregistreerd zijn of niet. Versus een metadatacataloog die zegt welke informatieproducten er allemaal bestaan met informatie over dienstverlening (Gouden Gids van informatiebronnen over dienstverleningen).

**Comment by GeraldineNolf** [comment]
@GeertThijs: aan wie kunnen we dat vragen? ;)

**Comment by GeertThijs** [comment]
@GeraldineNolf: Dat was ironisch zeker? Ik vraag het aan jou natuurlijk!

**Comment by GeraldineNolf** [comment]
Ik denk dat je basisregisters nodig hebt, waarin zaken op een unieke manier opgenomen worden. Als je zo'n register publiek ontsluit op een website, en je daarin zoeken mogelijk maakt, dan heb je een catalogus... Wat die zoekdienst en catalogus zelf op hun beurt ook weer 2 dienstverleningen maken richting gebruikers...

**Comment by GeraldineNolf** [comment]
Denk alleen dat niemand daar echt een antwoord op heeft... ? :)

**Comment by tthoeye** [comment]
Ik denk dat een Rule kan bestaan zonder dat daar noodzakelijk een PublicService aan gekoppeld is. Zeker wanneer we het vertalen als een "reglement" - zo komt b.v. ook een verkeersreglement in scope (zie bv https://stad.gent/sites/default/files/page/documents/Rommelmarkt_richtlijnen.pdf)

**Comment by GeertThijs** [comment]
Ik paste dit intussen al aan in het refactored CPSV-AP model in EA. Zie laatste comment bij #7. @tthoeye: de link in jouw voorbeeld is niet die van het verkeersreglement maar van de rommelmarkt.
